### PR TITLE
fixes a pacifism kill through crushing firelocks!

### PR DESCRIPTION
**Pull request by ghost**
15 minute fix from a friend who explained it in discord

### CODE REVIEW

**Comment by TheDreamweaver** [comment]
But what if

File: code/game/machinery/doors/firedoor.dm [hunk 1]
```diff
@@ -142,6 +142,17 @@
 	if(density)
 		open()
 	else
+		if(iscarbon(user)
```
Comment (ShizCalev) [inline code/game/machinery/doors/firedoor.dm]: )

Comment (ShizCalev) [inline code/game/machinery/doors/firedoor.dm]:
```
		if(iscarbon(user))
			var/mob/living/carbon/C = user
			if(C.has_trait(TRAIT_PACIFISM))
				var/T = get_turf(src)
				for(var/mob/living/L in T)
					if(L.stat != DEAD)
						to_chat(user, "<span class='notice'>You would hurt [L] by closing [src]!</span>")
						return
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by KorPhaeron** [comment]
Putting pacifist checks all over the codebase seems like a silly longterm plan

**Comment by ShizCalev** [comment]
new meta to find fakedeath liggers

**Comment by zaracka** [comment]
There's a lot of ways to bypass the pacifism check. Some are trivial and not really worth covering, I don't think this one is though. Although, one could argue that this is indirect harm and therefore okay? I dunno.

Examples: Exposing an area to space, injecting lethal reagents, using disarm intent to push someone into something deadly like a chasm, fire, etc.

File: code/game/machinery/doors/firedoor.dm [hunk 2]
```diff
@@ -142,6 +142,14 @@
 	if(density)
 		open()
 	else
+		if(iscarbon(user))
+			var/mob/living/carbon/C = user
+			if(C.has_trait(TRAIT_PACIFISM))
+				var/T = get_turf(src)
+				for(var/mob/living/L in T)
+					if((L.stat != DEAD) && !L.has_trait(TRAIT_FAKEDEATH))
```
Comment (ShizCalev) [inline code/game/machinery/doors/firedoor.dm]: i was a touch drunker when i typed this up, you can remove the fakedeath check. wouldn't really be a concern.

**Comment by zaracka** [comment]
I didn't mention this method in deadchat for the reason that I think it will be the go to method to kill someone as a pacifist. It's literally as easy as slipping someone (with water, soap, etc.) and crowbarring a firelock (which are everywhere). If the consensus is that it's creative/clever, then I guess keep it then.

Activating a fire alarm is a little more difficult and more in the feign-ignorance area.

**Comment by Armhulen** [comment]
@Barhandar i think turning on and off a fire alarm is indirect, but closing a door on someone to crush them isn't.

**Comment by epicmemename** [comment]
@Armhulen close this pr right now young man

**Comment by Rinoka** [comment]
yeah I support this PR. crowbar-killing someone is way too easy under pacifism

i'm fine with every other prevention method, this one just lets you crushspam firelocks

**Comment by iskyp** [comment]
I like how pacifism is the only divisive trait in this codebase

**Comment by Barhandar** [comment]
How _fast_ is it to perform, actually? If it's too fast, blocking only crowbaring on top of someone, and not fire-alarming, could work to prevent silent pax murder.

**Comment by Armhulen** [comment]
@Barhandar That's what that does. I think directly lowering the firelock onto someone shouldn't be a pacifist kill and from a gameplay standpoint firelocks are everywhere, but if you can stun them then run over to a fire alarm then sure have the kill i guess

**Pull request merged**